module/zfs/zpl_file.c

```diff
@@ -277,6 +278,7 @@ zpl_iter_read_common(struct kiocb *kiocb, const struct iovec *iovp,
 	    nr_segs, &kiocb->ki_pos, seg, filp->f_flags, cr, skip);
 	crfree(cr);
 
+	file_accessed(filp);
```

Comment:
We _may_ need to do the same thing for mmap if the VFS doesn't handle this for us with a dirty inode call.  All that's strictly required is atime be update sometime between mmap() and munmap().